kclvm/tools/src/query/override.rs

```diff
@@ -138,44 +274,107 @@ impl<'ctx> MutSelfMutWalker<'ctx> for OverrideTransformer {
+        // Loop all entries in the config expression and replace.
+        for (i, item) in config_expr.items.iter_mut().enumerate() {
+            // Compare each field of the config structure one by one.
+            // - `get_path_from_key` returns the real config key name.
+            // - `part` denotes the path entered by the user to be modified.
+            // For example, the real config node is `a: {b: c: {}}`. The path
+            // that needs to be modified is `a.b.c`, and its parts are ["a", "b", "c"].
+            if part == get_key_path(&item.node.key) {
+                match self.action {
+                    ast::OverrideAction::CreateOrUpdate => {
+                        // When the last part of the path is successfully recursively matched,
+                        // it indicates that the original value that needs to be overwritten
+                        // is successfully found, and the new value is used to overwrite it.
+                        // - `parts.len() == 1` denotes the path matches exactly.
+                        if parts.len() == 1 {
+                            self.override_value.set_pos(item.pos());
+                            // Override the node value.
+                            item.node.value = self.override_value.clone();
+                        } else {
+                            // Replace value recursively using the path composed by subsequent parts.
+                            //
+                            // The reason for using recursion instead of looping for path matching
+                            // is that rust cannot directly hold shared references to AST nodes
+                            // (ast::NodeRef<T> is a Box<T>), so recursive search is performed
+                            // directly on AST nodes.
+                            let config_expr = match &mut item.node.value.node {
+                                ast::Expr::Schema(schema_expr) => {
+                                    if let ast::Expr::Config(config_expr) =
+                                        &mut schema_expr.config.node
+                                    {
+                                        config_expr
+                                    } else {
+                                        continue;
+                                    }
+                                }
+                                ast::Expr::Config(config_expr) => config_expr,
+                                _ => {
+                                    continue;
+                                }
+                            };
+                            self.replace_config_with_path_parts(config_expr, &parts[1..]);
+                        }
+                    }
+                    ast::OverrideAction::Delete => {
+                        delete_index_set.insert(i);
+                    }
+                }
+            }
+        }
+        if !delete_index_set.is_empty() {
```

Comment:
It seems that nested case is not handled within deletion logic. 
Pls add more test case and see if we have a bug here.

Reply:
> It seems that nested case(a.b.c) is not handled within deletion logic. Pls add more test case and see if we have a bug here.

Have added more comments and fixed the bug.